tsfresh/feature_extraction/extraction.py

```diff
@@ -214,171 +190,112 @@ def _extract_features_per_kind(kind_to_df_map,
-                                          chunksize=defaults.CHUNKSIZE,
-                                          n_processes=defaults.N_PROCESSES, show_warnings=defaults.SHOW_WARNINGS,
-                                          disable_progressbar=defaults.DISABLE_PROGRESSBAR,
-                                          impute_function=defaults.IMPUTE_FUNCTION):
-    """
-    Parallelize the feature extraction per kind and per sample.
+    if len(result) != 0:
+        result = result.pivot("id", "variable", "value")
 
-    As the splitting of the dataframes per kind along column_id is quite costly, we settled for an async map in this
-    function. The result objects are temporarily stored in a fifo queue from which they can be retrieved in order
-    of submission.
+    return result
 
-    :param kind_to_df_map: The time series to compute the features for in our internal format
-    :type kind_to_df_map: dict of pandas.DataFrame
 
-    :param column_id: The name of the id column to group by.
-    :type column_id: str
+def _do_extraction_on_chunk(chunk, default_fc_parameters, kind_to_fc_parameters):
+    """
+    Main function of this module: use the feature calculators defined in the
+    default_fc_parameters or kind_to_fc_parameters parameters and extract all
+    features on the chunk.
 
-    :param column_value: The name for the column keeping the value itself.
-    :type column_value: str
+    The chunk consists of the chunk id, the chunk kind and the data (as a Series),
+    which is then converted to a numpy array - so a single time series.
 
-    :param default_fc_parameters: mapping from feature calculator names to parameters. Only those names
-           which are keys in this dict will be calculated. See the class:`ComprehensiveFCParameters` for
-           more information.
-    :type default_fc_parameters: dict
+    Returned is a list of the extracted features. Each one is a dictionary consisting of
+    { "variable": the feature name in the format <kind>__<feature>__<parameters>,
+      "value": the number value of the feature,
+      "id": the id of the chunk }
 
-    :param kind_to_fc_parameters: mapping from kind names to objects of the same type as the ones for
-            default_fc_parameters. If you put a kind as a key here, the fc_parameters
-            object (which is the value), will be used instead of the default_fc_parameters.
-    :type kind_to_fc_parameters: dict
+    The <parameters> are in the form described in :mod:`~tsfresh.utilities.string_manipulation`.
 
-    :param chunksize: The size of one chunk for the parallelisation
-    :type chunksize: None or int
+    :param chunk: A tuple of chunk_id, chunk_kind, data
+    :param default_fc_parameters: A dictionary of feature calculators.
+    :param kind_to_fc_parameters: A dictionary of fc_parameters for special kinds or None.
+    :return: A list of calculated features.
+    """
+    chunk_id, chunk_kind, data = chunk
```

Comment:
`df_melt` refers to the pandas method, that does exactly, what is done in the normalize function (see https://pandas.pydata.org/pandas-docs/stable/generated/pandas.melt.html).